cmd/admin-router.go

```diff
@@ -60,7 +60,7 @@ func registerAdminRouter(router *mux.Router, enableConfigOps bool) {
 
 		// Info operations
 		adminRouter.Methods(http.MethodGet).Path(adminVersion + "/info").HandlerFunc(gz(httpTraceAll(adminAPI.ServerInfoHandler)))
-		adminRouter.Methods(http.MethodGet).Path(adminVersion+"/inspect-data").HandlerFunc(httpTraceHdrs(adminAPI.InspectDataHandler)).Queries("volume", "{volume:.*}", "file", "{file:.*}")
+		adminRouter.Methods(http.MethodGet, http.MethodPost).Path(adminVersion + "/inspect-data").HandlerFunc(httpTraceAll(adminAPI.InspectDataHandler))
```

Comment:
> you are still not modifying anything on the server so a POST is not a correct http method.
That's PUT, POST is okay to be used to send data to the server for processing (https://www.rfc-editor.org/rfc/rfc7231#section-4.3.3)

> My suggestion is add this as a query param api=2 to differentiate
I can do that but I don't see benefits of doing that.

Basically, the change in this PR is:
- If the public key is not provided, follow the old mechanism which is sending the random key to mc to print it in the screen
- If the public key is provided, encrypt the random key with the public key and save it in a file in the archive

This is only a POST request because some load balancers restrict header value size to different values, though HTTP doesn't specify any limit, nginx restricts it to 4k by default, so this is a gray area

Reply:
what is the limit in query params? @vadmeste

Reply:
> what is the limit in query params? @vadmeste

There is no limit in RFC but all load balancers have its own limit, 2k, 4k, 8k, ..